dispatcher/brokers/pg_notify.py

```diff
@@ -97,8 +98,8 @@ def get_publish_channel(self, channel: Optional[str] = None) -> str:
     # --- asyncio connection methods ---
 
     async def aget_connection(self) -> psycopg.AsyncConnection:
-        "Return existing connection or create a new one"
-        if not self._async_connection:
+        # Check if the cached async connection is either None or closed.
+        if not self._async_connection or getattr(self._async_connection, "closed", 0) != 0:
```

Comment:
There's got to be a story behind `getattr(self._async_connection, "closed", 0)` here, and I'd like to hear it!